composite/token-allocation/src/adapter.ts

```diff
@@ -43,11 +43,14 @@ const inputParams = {
 const toValidAllocations = (allocations: TokenAllocations): TokenAllocations => {
   if (!allocations.every((t) => !!t.symbol))
     throw new AdapterError({ message: `Symbol not available for all tokens.`, statusCode: 400 })
-  return allocations.map((t) => ({
-    symbol: t.symbol.toUpperCase(),
-    decimals: t.decimals || DEFAULT_TOKEN_DECIMALS,
-    balance: t.balance || DEFAULT_TOKEN_BALANCE * 10 ** (t.decimals || DEFAULT_TOKEN_DECIMALS),
-  }))
+  return allocations.map((t) => {
+    const decimals = Number.isInteger(t.decimals) ? t.decimals : DEFAULT_TOKEN_DECIMALS
```

Comment:
In the case of synth, they are giving us the balance already in "units", so it would make sense to have an option to send `decimal: 0`.

Reply:
We can also add a check to avoid negative numbers. Both `decimals` and `balance` should be >= 0.

Consider creating three functions inside `toValidAllocations`:

- `_toValidSymbol`
- `_toValidDecimals`
- `_toValidBalance`